src/main/java/uk/gov/hmcts/darts/arm/service/impl/ArmRpoPollServiceImpl.java

```diff
@@ -0,0 +1,174 @@
+    private final ArmApiService armApiService;
+    private final ArmRpoService armRpoService;
+    private final UserIdentity userIdentity;
+    private final FileOperationService fileOperationService;
+    private final ArmDataManagementConfiguration armDataManagementConfiguration;
+
+    private List<File> tempProductionFiles;
+
+    @Override
+    public void pollArmRpo(boolean isManualRun) {
+        tempProductionFiles = new ArrayList<>();
+        try {
+            var armRpoExecutionDetailEntity = getArmRpoExecutionDetailEntity(isManualRun);
+            if (isNull(armRpoExecutionDetailEntity)) {
+                log.warn("Unable to find armRpoExecutionDetailEntity to poll");
+                return;
+            }
+            var bearerToken = armApiService.getArmBearerToken();
+            if (isNull(bearerToken)) {
+                log.warn("Unable to get bearer token to poll ARM RPO");
+                return;
+            }
+
+            var executionId = armRpoExecutionDetailEntity.getId();
+            var userAccount = userIdentity.getUserAccount();
+
+            // step to call ARM RPO API to get the extended searches by matter
+            log.debug("About to call getExtendedSearchesByMatter");
+            armRpoApi.getExtendedSearchesByMatter(bearerToken, executionId, userAccount);
+            // step to call ARM RPO API to get the master index field by record class schema
+            log.debug("About to call getMasterIndexFieldByRecordClassSchema");
+            List<MasterIndexFieldByRecordClassSchema> headerColumns = armRpoApi.getMasterIndexFieldByRecordClassSchema(
+                bearerToken, executionId,
+                ArmRpoHelper.getMasterIndexFieldByRecordClassSchemaSecondaryRpoState(),
+                userAccount);
+
+            // step to call ARM RPO API to create export based on search results table
+            log.debug("About to call createExportBasedOnSearchResultsTable");
+            var createExportBasedOnSearchResultsTable = armRpoApi.createExportBasedOnSearchResultsTable(bearerToken, executionId, headerColumns,
+                                                                                                        userAccount);
+            if (createExportBasedOnSearchResultsTable) {
+                // step to call ARM RPO API to get the extended productions by matter
+                log.debug("About to call getExtendedProductionsByMatter");
+                armRpoApi.getExtendedProductionsByMatter(bearerToken, executionId, userAccount);
+                // step to call ARM RPO API to get the production output files
+                log.debug("About to call getProductionOutputFiles");
+                var productionOutputFiles = armRpoApi.getProductionOutputFiles(bearerToken, executionId, userAccount);
+
+                for (var productionExportFileId : productionOutputFiles) {
+                    String productionExportFilename = generateProductionExportFilename(productionExportFileId);
```

Comment:
Just checking this is needed, as I don't think it's mentioned on the jira ticket.

Reply:
Ignore, I thought this was an API call, but it isn't.